### DISCUSSION TITLE
Question about `loss` in Gluon Estimator

### DISCUSSION THREAD

**Discussion by harupy** [apache/incubator-mxnet#19643] [Q&A]
In the current implementation, `loss` is monitored only when `self._train_metrics` is falsey (meaning `train_metrics` in `Estimator.__init__` is unspecified). I'm just curious if this is an intended behavior or not.

https://github.com/apache/incubator-mxnet/blob/4405906977e99c3328b0879a6dbe4ce56bf30d57/python/mxnet/gluon/contrib/estimator/estimator.py#L240-L246

**Comment by harupy** [comment]
In mxnet 1.5.x, `loss` is always monitored regardless of `self.train_metrics` (line 186  ~ 188).

https://github.com/apache/incubator-mxnet/blob/ec36dfe3af277c55c81646302e892895cb5908ac/python/mxnet/gluon/contrib/estimator/estimator.py#L180-L190

**Reply by szha** [reply]
cc @roywei
